### ISSUE TITLE
intel build no longer work with intel-mpi/2019.4 on Grizzly

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/MPAS-Model#785]
I've been trying to build with `ifort` from the Makefile and `intel-mpi`.  This used to work on Grizzly but doesn't seem to work anymore. The error when trying to build the PIO2 test program is:
```
ld: cannot find -lmpif90
```

However, if I change `mpif90`, `mpicc` and `mpicxx` to `mpiifort`, `mpiicc` and `mpicpc`, I am able to build successfully but I get errors at runtime:
```
[gr0250:32576] *** An error occurred in MPI_Info_dup
[gr0250:32576] *** reported by process [2785673216,0]
[gr0250:32576] *** on communicator MPI_COMM_WORLD
[gr0250:32576] *** MPI_ERR_INFO: invalid info object
[gr0250:32576] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[gr0250:32576] ***    and potentially your MPI job)
```

Additionally please enter this information if applicable:
 - Compiler and library stack:
```
module purge
module load git
module load friendly-testing
module load intel/19.0.4 intel-mpi/2019.4 hdf5-parallel/1.8.16 pnetcdf/1.11.2 \
    netcdf-h5parallel/4.7.3 mkl/2019.0.4
export NETCDF=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16
export NETCDFF=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16
export PNETCDF=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16
export PIO=/usr/projects/climate/xylar/libraries/grizzly/scorpio-1.1.5-intel
export I_MPI_CC=icc
export I_MPI_CXX=icpc
export I_MPI_F77=ifort
export I_MPI_F90=ifort
export CORE=ocean
export USE_PIO2=true
export AUTOCLEAN=true
export  HDF5_USE_FILE_LOCKING=FALSE
```
 - Test case used: `ocean/baroclinic_channel/10km/default`

**Comment by mattdturner** [comment]
@xylar Is this still an issue for you?  I was just able to successfully build and run (the full nightly suite for ocean) on Grizzly w/ Intel.  I did need to make the Makefile changes that you mentioned:
```
diff --git a/Makefile b/Makefile
index b4d0ea216..e080e4645 100644
--- a/Makefile
+++ b/Makefile
@@ -208,9 +208,9 @@ pgi-llnl:
 
 ifort:
        ( $(MAKE) all \
-       "FC_PARALLEL = mpif90" \
-       "CC_PARALLEL = mpicc" \
-       "CXX_PARALLEL = mpicxx" \
+       "FC_PARALLEL = mpiifort" \
+       "CC_PARALLEL = mpiicc" \
+       "CXX_PARALLEL = mpiicpc" \
        "FC_SERIAL = ifort" \
        "CC_SERIAL = icc" \
        "CXX_SERIAL = icpc" \
```

It looks like the only difference between my environment and yours is that I am using the `pio2` module found in `/usr/projects/climate/SHARED_CLIMATE/modulefiles/all/scorpio`.  My modules:
```
[13:08]mturner@gr-fe2 (ocean/develop):module list

Currently Loaded Modules:
  1) friendly-testing   4) hdf5-parallel/1.8.16      7) mkl/2019.0.4  
  2) intel/19.0.4       5) pnetcdf/1.11.2            8) pio2/1.10.1
  3) intel-mpi/2019.4   6) netcdf-h5parallel/4.7.3   9) git/2.21.0
```

**Comment by xylar** [comment]
Thanks, I'll try this out tomorrow. The fact that we have to edit the `Makefile` seems like it's not sustainable but it's a start if the suite at least works.

**Comment by xylar** [comment]
@mattdturner, thanks again!  I can confirm that things are working for me again.  I built the latest SCORPIO (1.1.6) as part of my testing.  Not sure if that made the difference or something changed on Grizzly or what, but glad to have things working.  I'll put in a PR to add `intel-mpi` as an option in the `Makefile`.

**Comment by xylar** [comment]
The following don't appear to be necessary:
```
export I_MPI_CC=icc
export I_MPI_CXX=icpc
export I_MPI_F77=ifort
export I_MPI_F90=ifort
```

**Comment by xylar** [comment]
This has been addressed via #816
